xla/service/gpu/backend_configs.proto

```diff
@@ -197,3 +197,18 @@ message CudnnfMHABackendConfig {
   // Is causal mask
   bool is_causal_mask = 21;
 }
+
+// Generic backend config for XLA:GPU
+message GpuBackendConfig {
```

Comment:
Would it clash with other backend configs we attach to instructions? E.g. using it to store tiling?

Reply:
+1. It might make sense to make GPU backend config the single entry point that potentially contains other existing backend config types as members similar to other backends, but that is a larger change.

Reply:
looks like any new config will overwrite the existing config so there's a good chance that the GpuBackendConfig will be lost or vice versa. I think @cezheng 's suggestion is a good long-term solution but it'd take significantly more effort. For this PR, can I put these 2 fields in metadata temporarily? I will add a TODO and work on a followup pr to refactor various gpu backend configs and have a single entry.

Reply:
Any comments on the metadata route? @cheshire @cezheng

Reply:
Metadata isn't great either, this is meant for, well, metadata. Maybe create new backend config to unify previous one + whatever data you need?

Reply:
fyi, i have been working on unifying all configs into a single GpuBackendConfig. still fixing some local test failures, but should have a new update soon.

Reply:
refactored all gpu backend configs into a single GpuBackendConfig